### PR TITLE
Add e2e tests for Number field

### PR DESCRIPTION
**Pull request by joerter**
@webteckie here are the tests for the number field. I also took took out the nth-child selectors for the Boolean field in adminUI.

### CODE REVIEW

File: test/e2e/adminUI/adminUI.js [hunk 1]
```diff
@@ -151,14 +151,14 @@ module.exports = {
...
 						},
 						fieldB: {
-							button: '.EditForm-container .field-type-boolean:nth-child(6) label button',
-							label: '.EditForm-container .field-type-boolean:nth-child(6) label span',
-							value: '.EditForm-container .field-type-boolean:nth-child(6) label input[name="fieldB"]'
+							button: '.EditForm-container .field-type-boolean[for="fieldB"] button',
+							label: '.EditForm-container .field-type-boolean[for="fieldB"] span',
+							value: '.EditForm-container .field-type-boolean[for="fieldB"] input[name="fieldB"]'
 						}
 					}
```
Comment (webteckie) [inline test/e2e/adminUI/adminUI.js]: minor comment here...@JedWatson want tests to eventually pass the same linting rules applied to keystone core, one of which is comma dangling.  Please add a trailing comma to to all JSON lines...see others for example...tnx

File: test/e2e/adminUI/adminUI.js [hunk 2]
```diff
@@ -294,9 +294,9 @@ module.exports = {
 							value: '.Modal-dialog .field-type-text[for="name"] input[name="name"]',
 						},
 						fieldA: {
-							button: '.Modal-dialog .field-type-boolean button',
-							label: '.Modal-dialog .field-type-boolean span',
-							value: '.Modal-dialog .field-type-boolean input[name="fieldA"]',
+							button: '.Modal-dialog .field-type-boolean[for="fieldA"] button',
+							label: '.Modal-dialog .field-type-boolean[for="fieldA"] span',
+							value: '.Modal-dialog .field-type-boolean[for="fieldA"] input[name="fieldA"]',
 						}
```
Comment (webteckie) [inline test/e2e/adminUI/adminUI.js]: ditto

File: test/e2e/adminUI/group005Fields/uiTest006NumberField.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		browser.expect.element(adminUI.cssSelector.initialModalView.fieldType.number.number.name.label)
+			.text.to.equal('Name');
+		browser.expect.element(adminUI.cssSelector.initialModalView.fieldType.number.number.name.value)
+			.to.be.visible;
+
+		browser.expect.element(adminUI.cssSelector.initialModalView.fieldType.number.number.fieldA.label)
+			.to.be.visible;
+		browser.expect.element(adminUI.cssSelector.initialModalView.fieldType.number.number.fieldA.label)
+			.text.to.equal('Field A');
+	}
```
Comment (webteckie) [inline test/e2e/adminUI/group005Fields/uiTest006NumberField.js]: can we also check here for the existence of the value field?

**Comment by webteckie** [comment]
@joerter love it!  Just some minor comments :-)  tnx

**Comment by joerter** [comment]
@webteckie awesome, I think I found all the instances of dangling commas let me know if there's any I missed.

File: test/e2e/adminUI/adminUI.js [hunk 4]
```diff
@@ -251,8 +251,8 @@ module.exports = {
 						fieldB: {
 							label: '.EditForm-container .field-type-number[for="fieldB"] label',
 							value: '.EditForm-container .field-type-number[for="fieldB"] input[name="fieldB"]'
```
Comment (webteckie) [inline test/e2e/adminUI/adminUI.js]: add one here

File: test/e2e/adminUI/adminUI.js [hunk 5]
```diff
@@ -164,8 +164,8 @@ module.exports = {
 							button: '.EditForm-container .field-type-boolean[for="fieldB"] button',
 							label: '.EditForm-container .field-type-boolean[for="fieldB"] span',
 							value: '.EditForm-container .field-type-boolean[for="fieldB"] input[name="fieldB"]'
```
Comment (webteckie) [inline test/e2e/adminUI/adminUI.js]: add one here

**Comment by webteckie** [comment]
Missed a couple but one of them was my bad :-)  We should probably run lint on this at some point to cover all the basis...but for now eyeballing it is fine :-)  Also I had added a couple more comments in the original review...see above.

**Comment by joerter** [comment]
@webteckie No problem, got em!

**Comment by webteckie** [comment]
How about the comment to check for the presence of the number field A value in the initial modal?  Does it make sense to do so in the ui test?

File: test/e2e/adminUI/group005Fields/uiTest006NumberField.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+		browser.expect.element(adminUI.cssSelector.initialModalView.fieldType.number.number.name.label)
+			.text.to.equal('Name');
+		browser.expect.element(adminUI.cssSelector.initialModalView.fieldType.number.number.name.value)
+			.to.be.visible;
+
+		browser.expect.element(adminUI.cssSelector.initialModalView.fieldType.number.number.fieldA.label)
+			.to.be.visible;
+		browser.expect.element(adminUI.cssSelector.initialModalView.fieldType.number.number.fieldA.label)
+			.text.to.equal('Field A');
+		browser.expect.element(adminUI.cssSelector.initialModalView.fieldType.number.number.fieldA.value)
```
Comment (joerter) [inline test/e2e/adminUI/group005Fields/uiTest006NumberField.js]: Do you mean this check here?

File: test/e2e/adminUI/group005Fields/uiTest006NumberField.js [hunk 7]
```diff
@@ -37,5 +37,7 @@ module.exports = {
 			.to.be.visible;
 		browser.expect.element(adminUI.cssSelector.initialModalView.fieldType.number.number.fieldA.label)
 			.text.to.equal('Field A');
-	}
+		browser.expect.element(adminUI.cssSelector.initialModalView.fieldType.number.number.fieldA.value)
```
Comment (joerter) [inline test/e2e/adminUI/group005Fields/uiTest006NumberField.js]: Do you mean this check here?

Comment (joerter) [inline test/e2e/adminUI/group005Fields/uiTest006NumberField.js]: Do you mean this check here?

**Comment by webteckie** [comment]
ah, cool, did realize you had done it...merged!  which one you want to work on next?  :-)  tnx

**Comment by joerter** [comment]
No problem! Let's see... how about text? Any unique cases I should test for with that?

**Comment by webteckie** [comment]
No that I can think of...I think it just a string.  It's yours :-)  btw, the first pass we are just doing basic tests.  Another pass will test other field options as well as more in depth coverage of filtering, etc.  btw, I've also been thinking of restructuring adminUI.js since it's becoming a large module.  Somehow try to isolate field selector config into its own per field file.  Mulling it over...if you have any suggestions let's hear them :-)

**Comment by joerter** [comment]
Yeah I was looking at the nightwatch documentation today and was wondering what you think of page objects? If we had separate page objects for the different views we have now under `cssSelector` it might make things easier because we could add in commands to do things like log in / out, click on plus icons for fields, etc.

**Comment by webteckie** [comment]
Yeah, let's refactor using that pattern before we get deeper.  You want to take a stab at it?  If not, let me know and I will.   Thanks!

**Comment by webteckie** [comment]
Actually we can both attack this in parallel.   We just got to leave adminUI.js alone for now until we're done creating all the page objects and refactoring the tests to use those page objects instead of adminUI.js.  But let's just do the signin tests first so that we set the pattern and file structure down.

**Comment by joerter** [comment]
Sounds good! Going in parallel that way makes a lot of sense. I'll take a look tomorrow.

**Comment by webteckie** [comment]
@joerter when you get a chance head over to the slack design channel :-)

**Pull request merged**